### PR TITLE
T9f2-14 create and update mongoose group schema 

### PR DESCRIPTION
**Pull request by adityaputtigampala**

### CODE REVIEW

File: mongoose/groups/GroupSchema.ts [hunk 1]
```diff
@@ -0,0 +1,14 @@
+/**
+ * @file Implements mongoose schema for groups.
+ */
+ import mongoose, {Schema} from "mongoose";
+ import Group from "../../models/group/Group"
```
Comment (mitchneides) [inline mongoose/groups/GroupSchema.ts]: needs ; at end of line 5

File: mongoose/groups/GroupSchema.ts [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+ 
+ const GroupSchema = new mongoose.Schema<Group>({
+     members: {type:[]},
+     createdOn: {type: Date, default: Date.now()},
+     admin: {type:[]},
+     groupName: {type: String},
+     description: {type: String}
+ }, {collection: "groups"});
+ export default GroupSchema
```
Comment (mitchneides) [inline mongoose/groups/GroupSchema.ts]: Also needs ; at end of 14

File: mongoose/groups/GroupSchema.ts [hunk 3]
```diff
@@ -0,0 +1,14 @@
+/**
+ * @file Implements mongoose schema for groups.
+ */
+ import mongoose, {Schema} from "mongoose";
+ import Group from "../../models/group/Group"
+ 
+ const GroupSchema = new mongoose.Schema<Group>({
+     members: {type:[]},
```
Comment (mitchneides) [inline mongoose/groups/GroupSchema.ts:8]: For lines 8 and 10 do we need to explicitly define the array type?

Comment (adityaputtigampala) [reply mongoose/groups/GroupSchema.ts:8]: i tried doing something like the below for the array of strings, but im getting an error. I'm thinking of leaving it generic for now to see if it updates correctly in mongo when we test

https://stackoverflow.com/questions/45780272/how-to-define-an-array-of-strings-in-typescript-interface

**Comment by adityaputtigampala** [comment]
added colons to the feature branch

File: mongoose/groups/GroupSchema.ts (hunk 3 above)
Comment (mitchneides) [reply mongoose/groups/GroupSchema.ts:8]: Fair enough so yeah let's just keep it in mind as we progress to the next stages.
If we run into issues then we'll come back to check this

Comment (mitchneides) [reply mongoose/groups/GroupSchema.ts:8]: (unresolved it so others can check/give their feedback)

**Comment by mitchneides** [comment]
Unless the other reviewers have an idea for defining the array type (see discussion above), I'm good to merge this and just keep that in mind as we continue developing this feature

**Comment by Romangino** [comment]
Looks fine to me. We'll see if the type defined becomes an issue

**Comment by Kylefairbairn** [comment]
Looks fine to me and I agree with everyone else about the type error.

**Comment by spencerbelsky** [comment]
Looks good to me as well. We can keep tabs on this type and make changes as necessary

**Pull request merged**